lib/src/models/image/cover.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:dart_mappable/dart_mappable.dart';
+
+import 'cover_thumbnail.dart';
+import 'image.dart';
+import 'image_with_known_dimensions.dart';
+
+part 'cover.mapper.dart';
+
+
+/// Custom mapper for [Cover].
+class CoverHook extends MappingHook {
+  const CoverHook();
+
+  @override
+  Cover beforeDecode(dynamic value) {
+    if (value is! Map<String, dynamic>)
+      throw MapperException.unexpectedType(value.runtimeType, Cover, 'Map<String, dynamic>');
+
+    final image = Image.fromValue({
+      ...value,
+      'id': 0,
+    });
+    return Cover(
+      media: image.media, 
+      type: image.type, 
+      width: image.width!, 
+      height: image.height!,
+    );
+  }
+
+  @override
+  dynamic beforeEncode(dynamic value) => 
+    ImageMapper.container.toMap(value); 
+}
+
+/// Cover.
+@MappableClass(
+  hook: CoverHook(),
+)
+class Cover extends Image with ImageWithKnownDimensions, CoverMappable {
+  Cover({
+    required super.media,
+    required super.type,
+    required int width,
+    required int height,
+  }) : super(
+    id: 0, 
+    width: width, 
+    height: height,
```

Comment:
Why not `super.width` and `super.height`?

Reply:
`super.width` and `super.height` have `int?` types.
But we want to force `int` here.

Reply:
Ok then